File: ruby/lib/agentic_commerce_protocol/models/card_funding_type.rb
# frozen_string_literal: true
# This file is @generated
module AgenticCommerceProtocol
  class CardFundingType
    CREDIT = "credit".freeze
    DEBIT = "debit".freeze
    PREPAID = "prepaid".freeze

    def self.all_vars
      @all_vars ||= [CREDIT, DEBIT, PREPAID].freeze
    end

    def initialize(value)
      unless CardFundingType.all_vars.include?(value)
        raise "Invalid ENUM value '#{value}' for class #CardFundingType"
      end

      @value = value
    end

    def self.deserialize(value)
      return value if CardFundingType.all_vars.include?(value)
      raise "Invalid ENUM value '#{value}' for class #CardFundingType"
    end

    def serialize
      @value
    end
  end
end

File: java/src/main/java/com/svix/agentic_commerce_protocol/CardFundingType.java
// This file is @generated
package com.svix.agentic_commerce_protocol;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.agentic_commerce_protocol.Utils.ToQueryParam;

public enum CardFundingType implements ToQueryParam {
    CREDIT("credit"),
    DEBIT("debit"),
    PREPAID("prepaid");
    private final String value;

    CardFundingType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}
